### DISCUSSION TITLE
Swanctl - Windows asks for password twice

### DISCUSSION THREAD

**Discussion by aka-woolf** [strongswan/strongswan#1326] [Q&A] [answered]
Hi!

I'm using **strongSwan 6.0dr16** with swanclt and I have a problem with Windows native VPN client: it asks for the password the second time when we connect to the VPN. I know that _eap-identity_ plugin should be loaded, at least it works for ipsec config, but even though I see it among the loaded plugins on swanctl too, it doesn't seem to help. I also noticed that the list of loaded plugins is different when usind _swanctl --help_ and _swanctl --status_

Config:
```
authorities {
        ca-vpn-root {
                cacert = /etc/easyrsa/pki/ca.crt
        }
}
connections {
	conn-be-srv-2 {
		version = 2             # IKEv2
        	encap = yes
        	unique = replace        # Use unique IDs
        	dpd_delay = 20s         # Keepalive interval
        	keyingtries = 3         # Initiation attempts number
        	local_addrs = 159.223.239.251           # Local interface IP
        	remote_addrs = %any     # Allow connections from any IP
        	proposals = aes256gcm16-sha384-ecp384           #Phase 1 proposals
        	send_certreq = no
        	send_cert = always
        	rekey_time = 24h
        	pools = pool-be-2               # Assign static IP to Backend Server 1
        	local {                 # Server authN
                	auth = pubkey
                	certs = /etc/easyrsa/pki/issued/vpn-server.crt
                	id = 159.223.239.251
        	}
	        remote {
                	auth = eap-mschapv2     # Backend Server authN
               		eap_id = id-be-srv-2    # Backend Server EAP ID
        	}
        	children {                      # Phase 2 settings
        	        ipsec-be-2 {
        	                local_ts  = 10.10.10.0/24         # Local traffic selector
#       	                local_ts = 1.0.0.0/16
        	                esp_proposals = aes256gcm16     # Phase 2 proposals
        	                rekey_time = 4h                 # Phase 2 lifetime
        	        }
        	}
}
secrets {
        eap-be-2 {
                id = id-be-srv-2
                secret = XXXXXXXXX
        }
}
# Pools with static IPs:
pools {
        pool-be-2 {
                addrs = 10.0.0.2/32
        }
}
```

Connection:
```
root@vpn-server:/etc/swanctl# swanctl -L
conn-be-srv-2: IKEv2, no reauthentication, rekeying every 86400s, dpd delay 20s
  local:  159.223.239.251
  remote: %any
  local public key authentication:
    id: 159.223.239.251
    certs: CN=vpn-server
  remote EAP_MSCHAPV2 authentication:
    eap_id: id-be-srv-2
  ipsec-be-2: TUNNEL, rekeying every 14400s, dpd action is none
    local:  10.10.10.0/24
    remote: dynamic
```

Status:
```
root@vpn-server:/etc# swanctl -h
strongSwan 6.0dr16 swanctl
loaded plugins: aes sha2 sha1 md4 random nonce x509 revocation constraints pubkey pem openssl

root@vpn-server:/etc# swanctl -S
uptime: 45 seconds, since Oct 06 09:27:25 2022
worker threads: 16 total, 10 idle, working: 5/0/1/0
job queues: 0/0/0/0
jobs scheduled: 3
IKE_SAs: 1 total, 0 half-open
mallinfo: sbrk 2113536, mmap 0, used 1482960, free 630576
loaded plugins: charon-systemd aes sha2 sha1 md4 random nonce x509 revocation constraints pubkey pem openssl attr kernel-netlink resolve socket-default stroke vici updown eap-identity eap-md5 eap-mschapv2 eap-tls counters
```

Log:
```
Oct 06 08:54:40 vpn-server charon-systemd[649306]: received packet: from <Client-IP>[518] to <Server-IP>[500] (376 bytes)
Oct 06 08:54:40 vpn-server charon-systemd[649306]: parsed IKE_SA_INIT request 0 [ SA KE No N(FRAG_SUP) N(NATD_S_IP) N(NATD_D_IP) V V V V ]
Oct 06 08:54:40 vpn-server charon-systemd[649306]: received MS NT5 ISAKMPOAKLEY v9 vendor ID
Oct 06 08:54:40 vpn-server charon-systemd[649306]: received MS-Negotiation Discovery Capable vendor ID
Oct 06 08:54:40 vpn-server charon-systemd[649306]: received Vid-Initial-Contact vendor ID
Oct 06 08:54:40 vpn-server charon-systemd[649306]: received unknown vendor ID: 01:52:8b:bb:c0:06:96:12:18:49:ab:9a:1c:5b:2a:51:00:00:00:02
Oct 06 08:54:40 vpn-server charon-systemd[649306]: <Client-IP> is initiating an IKE_SA
Oct 06 08:54:40 vpn-server charon-systemd[649306]: selected proposal: IKE:AES_GCM_16_256/PRF_HMAC_SHA2_384/ECP_384
Oct 06 08:54:40 vpn-server charon-systemd[649306]: remote host is behind NAT
Oct 06 08:54:40 vpn-server charon-systemd[649306]: generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(CHDLESS_SUP) N(MULT_AUTH) ]
Oct 06 08:54:40 vpn-server charon-systemd[649306]: sending packet: from <Server-IP>[500] to <Client-IP>[518] (288 bytes)
Oct 06 08:54:40 vpn-server charon-systemd[649306]: received packet: from <Client-IP>[8843] to <Server-IP>[4500] (572 bytes)
Oct 06 08:54:40 vpn-server charon-systemd[649306]: parsed IKE_AUTH request 1 [ EF(1/3) ]
Oct 06 08:54:40 vpn-server charon-systemd[649306]: received fragment #1 of 3, waiting for complete IKE message
Oct 06 08:54:40 vpn-server charon-systemd[649306]: received packet: from <Client-IP>[8843] to <Server-IP>[4500] (572 bytes)
Oct 06 08:54:40 vpn-server charon-systemd[649306]: parsed IKE_AUTH request 1 [ EF(2/3) ]
Oct 06 08:54:40 vpn-server charon-systemd[649306]: received fragment #2 of 3, waiting for complete IKE message
Oct 06 08:54:40 vpn-server charon-systemd[649306]: received packet: from <Client-IP>[8843] to <Server-IP>[4500] (428 bytes)
Oct 06 08:54:40 vpn-server charon-systemd[649306]: parsed IKE_AUTH request 1 [ EF(3/3) ]
Oct 06 08:54:40 vpn-server charon-systemd[649306]: received fragment #3 of 3, reassembled fragmented IKE message (1402 bytes)
Oct 06 08:54:40 vpn-server charon-systemd[649306]: parsed IKE_AUTH request 1 [ IDi CERTREQ N(MOBIKE_SUP) CPRQ(ADDR DNS NBNS SRV ADDR6 DNS6 SRV6) SA TSi TSr ]
Oct 06 08:54:40 vpn-server charon-systemd[649306]: received cert request for "CN=VPN Root CA"
Oct 06 08:54:40 vpn-server charon-systemd[649306]: received 55 cert requests for an unknown ca
Oct 06 08:54:40 vpn-server charon-systemd[649306]: looking for peer configs matching <Server-IP>[%any]...<Client-IP>[172.16.4.102]
Oct 06 08:54:40 vpn-server charon-systemd[649306]: selected peer config 'conn-be-srv-2'
Oct 06 08:54:40 vpn-server charon-systemd[649306]: using configured EAP-Identity id-be-srv-2
Oct 06 08:54:40 vpn-server charon-systemd[649306]: initiating EAP_MSCHAPV2 method (id 0x64)
Oct 06 08:54:40 vpn-server charon-systemd[649306]: peer supports MOBIKE
Oct 06 08:54:40 vpn-server charon-systemd[649306]: authentication of '<Server-IP>' (myself) with RSA signature successful
Oct 06 08:54:40 vpn-server charon-systemd[649306]: sending end entity cert "CN=vpn-server"
Oct 06 08:54:40 vpn-server charon-systemd[649306]: generating IKE_AUTH response 1 [ IDr CERT AUTH EAP/REQ/MSCHAPV2 ]
Oct 06 08:54:40 vpn-server charon-systemd[649306]: splitting IKE message (1256 bytes) into 2 fragments
Oct 06 08:54:40 vpn-server charon-systemd[649306]: generating IKE_AUTH response 1 [ EF(1/2) ]
Oct 06 08:54:40 vpn-server charon-systemd[649306]: generating IKE_AUTH response 1 [ EF(2/2) ]
Oct 06 08:54:40 vpn-server charon-systemd[649306]: sending packet: from <Server-IP>[4500] to <Client-IP>[8843] (1248 bytes)
Oct 06 08:54:40 vpn-server charon-systemd[649306]: sending packet: from <Server-IP>[4500] to <Client-IP>[8843] (73 bytes)
Oct 06 08:54:40 vpn-server charon-systemd[649306]: received packet: from <Client-IP>[8843] to <Server-IP>[4500] (136 bytes)
Oct 06 08:54:40 vpn-server charon-systemd[649306]: parsed IKE_AUTH request 2 [ EAP/RES/MSCHAPV2 ]
Oct 06 08:54:40 vpn-server charon-systemd[649306]: EAP-MS-CHAPv2 username: '%any'
Oct 06 08:54:40 vpn-server charon-systemd[649306]: no EAP key found for hosts '%any' - '%any'
Oct 06 08:54:40 vpn-server charon-systemd[649306]: EAP-MS-CHAPv2 verification failed, retry (1)
Oct 06 08:54:41 vpn-server charon-systemd[649306]: ignoring request with ID 2, already processing
Oct 06 08:54:42 vpn-server charon-systemd[649306]: generating IKE_AUTH response 2 [ EAP/REQ/MSCHAPV2 ]
Oct 06 08:54:42 vpn-server charon-systemd[649306]: sending packet: from <Server-IP>[4500] to <Client-IP>[8843] (114 bytes)
Oct 06 08:54:42 vpn-server charon-systemd[649306]: received packet: from <Client-IP>[8843] to <Server-IP>[4500] (136 bytes)
Oct 06 08:54:42 vpn-server charon-systemd[649306]: received retransmit of request with ID 2, retransmitting response
Oct 06 08:54:42 vpn-server charon-systemd[649306]: sending packet: from <Server-IP>[4500] to <Client-IP>[8843] (114 bytes)
Oct 06 08:54:45 vpn-server charon-systemd[649306]: received packet: from <Client-IP>[8843] to <Server-IP>[4500] (136 bytes)
Oct 06 08:54:45 vpn-server charon-systemd[649306]: parsed IKE_AUTH request 3 [ EAP/RES/MSCHAPV2 ]
Oct 06 08:54:45 vpn-server charon-systemd[649306]: generating IKE_AUTH response 3 [ EAP/REQ/MSCHAPV2 ]
Oct 06 08:54:45 vpn-server charon-systemd[649306]: sending packet: from <Server-IP>[4500] to <Client-IP>[8843] (134 bytes)
Oct 06 08:54:45 vpn-server charon-systemd[649306]: received packet: from <Client-IP>[8843] to <Server-IP>[4500] (72 bytes)
Oct 06 08:54:45 vpn-server charon-systemd[649306]: parsed IKE_AUTH request 4 [ EAP/RES/MSCHAPV2 ]
Oct 06 08:54:45 vpn-server charon-systemd[649306]: EAP method EAP_MSCHAPV2 succeeded, MSK established
Oct 06 08:54:45 vpn-server charon-systemd[649306]: generating IKE_AUTH response 4 [ EAP/SUCC ]
Oct 06 08:54:45 vpn-server charon-systemd[649306]: sending packet: from <Server-IP>[4500] to <Client-IP>[8843] (65 bytes)
Oct 06 08:54:45 vpn-server charon-systemd[649306]: received packet: from <Client-IP>[8843] to <Server-IP>[4500] (120 bytes)
Oct 06 08:54:45 vpn-server charon-systemd[649306]: parsed IKE_AUTH request 5 [ AUTH ]
Oct 06 08:54:45 vpn-server charon-systemd[649306]: authentication of '172.16.4.102' with EAP successful
Oct 06 08:54:45 vpn-server charon-systemd[649306]: authentication of '<Server-IP>' (myself) with EAP
Oct 06 08:54:45 vpn-server charon-systemd[649306]: IKE_SA conn-be-srv-2[2] established between <Server-IP>[<Server-IP>]...<Client-IP>[172.16.4.102]
Oct 06 08:54:45 vpn-server charon-systemd[649306]: scheduling rekeying in 78180s
Oct 06 08:54:45 vpn-server charon-systemd[649306]: maximum IKE_SA lifetime 86820s
Oct 06 08:54:45 vpn-server charon-systemd[649306]: peer requested virtual IP %any
Oct 06 08:54:45 vpn-server charon-systemd[649306]: assigning new lease to 'id-be-srv-2'
Oct 06 08:54:45 vpn-server charon-systemd[649306]: assigning virtual IP 10.0.0.2 to peer 'id-be-srv-2'
Oct 06 08:54:45 vpn-server charon-systemd[649306]: peer requested virtual IP %any6
Oct 06 08:54:45 vpn-server charon-systemd[649306]: no virtual IP found for %any6 requested by 'id-be-srv-2'
Oct 06 08:54:45 vpn-server charon-systemd[649306]: selected proposal: ESP:AES_GCM_16_256/NO_EXT_SEQ
Oct 06 08:54:45 vpn-server charon-systemd[649306]: CHILD_SA ipsec-be-2{3} established with SPIs ca857949_i cb7c6456_o and TS 10.10.10.0/24 === 10.0.0.2/32
Oct 06 08:54:45 vpn-server charon-systemd[649306]: generating IKE_AUTH response 5 [ AUTH CPRP(ADDR) SA TSi TSr N(MOBIKE_SUP) N(ADD_4_ADDR) N(ADD_4_ADDR) ]
Oct 06 08:54:45 vpn-server charon-systemd[649306]: sending packet: from <Server-IP>[4500] to <Client-IP>[8843] (245 bytes)
```

Any hints would be appreciated!

Thanks!

**Comment by tobiasbrunner** [comment, marked as answer]
> it asks for the password the second time when we connect to the VPN.

What do you mean? Why shouldn't it ask for a password?

> I know that _eap-identity_ plugin should be loaded, at least it works for ipsec config, but even though I see it among the loaded plugins on swanctl too, it doesn't seem to help.

That's because this

> eap_id = id-be-srv-2    # Backend Server EAP ID

is not how you request an EAP-Identity from the client. That requires `eap_id = %any`. If you want to match configs against EAP identities, that requires a workaround (see e.g. [this one for ipsec.conf](https://serverfault.com/a/908194/95913), which can be adapted similarly for swanctl.conf). Did you use that workaround with ipsec.conf?

> I also noticed that the list of loaded plugins is different when usind _swanctl --help_ and _swanctl --status_

Yes, the former lists the plugins loaded by `swanctl` itself e.g. to parse certificates/keys, the latter those loaded by the IKE daemon as reported via vici.

**Comment by aka-woolf** [comment]
Hi @tobiasbrunner , thanks for the quick reply!

By "asking the second time" I meant that after providing the username and password pair the first time the VPN connects, I'm asked for just for the password the second time. When the credentials are remembered I'm asked for just the password again each time I connect to the VPN.

But I followed your link and the trick with adding a group check for connection helped! Now the VPN connects without issues! I had to modify the config files though and now they look like this:

swanctl.conf:
```
authorities {
        ca-vpn-root {
                cacert = /etc/easyrsa/pki/ca.crt
        }
}
connections {
        # Dummy connection to trigger EAP_IDENTITY method:
        conn-eap-init {
                include conf.d/conn-default-ra.conf
                include conf.d/conn-eap-init.conf
        }
        # Backend Server 2:
        conn-be-srv-2 {
                include conf.d/conn-default-ra.conf
                include conf.d/conn-be-srv-2.conf
        }
}
secrets {
        eap-be-2 {
                id = id-be-srv-2
                secret = XXXXXXXXX
        }
}
# Pools with static IPs:
pools {
        pool-be-2 {
                addrs = 10.0.0.2/32
        }
}
```
conn-default-ra.conf:
```
        version = 2             # IKEv2
        encap = yes
        unique = replace        # Use unique IDs
        dpd_delay = 20s         # Keepalive interval
        keyingtries = 3         # Initiation attempts number
        local_addrs = 159.223.239.251           # Local interface IP
        remote_addrs = %any     # Allow connections from any IP
        proposals = aes256gcm16-sha384-ecp384           #Phase 1 proposals
        send_certreq = no
        send_cert = always
        rekey_time = 24h
        local {                 # Server authN
                auth = pubkey
                certs = /etc/easyrsa/pki/issued/vpn-server.crt
                id = 159.223.239.251
        }
```

conn-eap-init.conf:
```
        children {                      # Phase 2 settings
                ipsec-eap-init {

                }
        }
        remote {
                auth = eap-mschapv2     # Backend Server authN
                groups = group-auth-fake
                eap_id = %any   # Backend Server EAP ID
        }
```

conn-be-srv-2.conf:
```
        pools = pool-be-2               # Assign static IP to Backend Server 1
        children {                      # Phase 2 settings
                ipsec-be-2 {
                        local_ts  = 10.10.10.0/24         # Local traffic selector
#                       local_ts = 1.0.0.0/16
                        esp_proposals = aes256gcm16     # Phase 2 proposals
                        rekey_time = 4h                 # Phase 2 lifetime
                }
        }
        remote {
                auth = eap-mschapv2     # Backend Server authN
                eap_id = id-be-srv-2    # Backend Server EAP ID
        }
```

And the connection log looks like this:
<pre>
Oct 06 12:02:53 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received packet: from 95.179.217.105[4500] to 159.223.239.251[4500] (57 bytes)
Oct 06 12:02:53 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: parsed INFORMATIONAL request 68 [ ]
Oct 06 12:02:53 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: generating INFORMATIONAL response 68 [ ]
Oct 06 12:02:53 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: sending packet: from 159.223.239.251[4500] to 95.179.217.105[4500] (57 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received packet: from 31.173.91.25[515] to 159.223.239.251[500] (376 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: parsed IKE_SA_INIT request 0 [ SA KE No N(FRAG_SUP) N(NATD_S_IP) N(NATD_D_IP) V V V V ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received MS NT5 ISAKMPOAKLEY v9 vendor ID
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received MS-Negotiation Discovery Capable vendor ID
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received Vid-Initial-Contact vendor ID
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received unknown vendor ID: 01:52:8b:bb:c0:06:96:12:18:49:ab:9a:1c:5b:2a:51:00:00:00:02
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: 31.173.91.25 is initiating an IKE_SA
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: selected proposal: IKE:AES_GCM_16_256/PRF_HMAC_SHA2_384/ECP_384
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: remote host is behind NAT
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(CHDLESS_SUP) N(MULT_AUTH) ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: sending packet: from 159.223.239.251[500] to 31.173.91.25[515] (288 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received packet: from 31.173.91.25[9122] to 159.223.239.251[4500] (572 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: parsed IKE_AUTH request 1 [ EF(1/3) ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received fragment #1 of 3, waiting for complete IKE message
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received packet: from 31.173.91.25[9122] to 159.223.239.251[4500] (572 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: parsed IKE_AUTH request 1 [ EF(2/3) ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received fragment #2 of 3, waiting for complete IKE message
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received packet: from 31.173.91.25[9122] to 159.223.239.251[4500] (428 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: parsed IKE_AUTH request 1 [ EF(3/3) ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received fragment #3 of 3, reassembled fragmented IKE message (1402 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: parsed IKE_AUTH request 1 [ IDi CERTREQ N(MOBIKE_SUP) CPRQ(ADDR DNS NBNS SRV ADDR6 DNS6 SRV6) SA TSi TSr ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received cert request for "CN=VPN Root CA"
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received 55 cert requests for an unknown ca
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: looking for peer configs matching 159.223.239.251[%any]...31.173.91.25[172.16.4.102]
<b>Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: selected peer config 'conn-eap-init'</b>
<b>Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: initiating EAP_IDENTITY method</b>(id 0x00)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: peer supports MOBIKE
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: authentication of '159.223.239.251' (myself) with RSA signature successful
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: sending end entity cert "CN=vpn-server"
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: generating IKE_AUTH response 1 [ IDr CERT AUTH EAP/REQ/ID ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: sending packet: from 159.223.239.251[4500] to 31.173.91.25[9122] (1225 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received packet: from 31.173.91.25[9122] to 159.223.239.251[4500] (88 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: parsed IKE_AUTH request 2 [ EAP/RES/ID ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received EAP identity 'id-be-srv-2'
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: initiating EAP_MSCHAPV2 method (id 0x36)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: generating IKE_AUTH response 2 [ EAP/REQ/MSCHAPV2 ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: sending packet: from 159.223.239.251[4500] to 31.173.91.25[9122] (97 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received packet: from 31.173.91.25[9122] to 159.223.239.251[4500] (136 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: parsed IKE_AUTH request 3 [ EAP/RES/MSCHAPV2 ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: generating IKE_AUTH response 3 [ EAP/REQ/MSCHAPV2 ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: sending packet: from 159.223.239.251[4500] to 31.173.91.25[9122] (134 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received packet: from 31.173.91.25[9122] to 159.223.239.251[4500] (72 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: parsed IKE_AUTH request 4 [ EAP/RES/MSCHAPV2 ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: EAP method EAP_MSCHAPV2 succeeded, MSK established
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: generating IKE_AUTH response 4 [ EAP/SUCC ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: sending packet: from 159.223.239.251[4500] to 31.173.91.25[9122] (65 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: received packet: from 31.173.91.25[9122] to 159.223.239.251[4500] (120 bytes)
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: parsed IKE_AUTH request 5 [ AUTH ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: authentication of '172.16.4.102' with EAP successful
<b>Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: constraint check failed: group membership to 'group-auth-fake' required</b>
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: selected peer config 'conn-eap-init' unacceptable: non-matching authentication done
<b>Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: switching to peer config 'conn-be-srv-2</b>
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: authentication of '159.223.239.251' (myself) with EAP
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: IKE_SA conn-be-srv-2[3] established between 159.223.239.251[159.223.239.251]...31.173.91.25[172.16.4.102]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: scheduling rekeying in 82968s
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: maximum IKE_SA lifetime 91608s
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: peer requested virtual IP %any
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: reassigning offline lease to 'id-be-srv-2'
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: assigning virtual IP 10.0.0.2 to peer 'id-be-srv-2'
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: peer requested virtual IP %any6
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: no virtual IP found for %any6 requested by 'id-be-srv-2'
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: selected proposal: ESP:AES_GCM_16_256/NO_EXT_SEQ
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: CHILD_SA ipsec-be-2{4} established with SPIs c796abc3_i d7b814d4_o and TS 10.10.10.0/24 === 10.0.0.2/32
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: generating IKE_AUTH response 5 [ AUTH CPRP(ADDR) SA TSi TSr N(MOBIKE_SUP) N(ADD_4_ADDR) N(ADD_4_ADDR) ]
Oct 06 12:02:54 digitalocean-ubuntu-ams3-s-1vcpu-2gb charon-systemd[651646]: sending packet: from 159.223.239.251[4500] to 31.173.91.25[9122] (245 bytes)
<pre>

**Reply by tobiasbrunner** [reply]
Great. By the way, you could define this in a single config file via section references (e.g. reference the actual connection from the `conn-eap-init` connection, the referenced section can be later in the file, and just set/override the required settings):

```
connections {
        # Dummy connection to trigger EAP_IDENTITY method:
        conn-eap-init : connections.conn-be-srv-2 {
                remote {
                        groups = group-auth-fake
                        eap_id = %any
                }
        }
        conn-be-srv-2 {
...
```

**Reply by aka-woolf** [reply]
Thanks for the hint! I must have overlooked this feature in the documentation.
